### DISCUSSION TITLE
Apache Camel installation issue  " PKIX path building failed"

### DISCUSSION THREAD

**Discussion by nmvprasad** [jbangdev/jbang#1702] [Q&A]
getting this error on my windows 10 when i try to install Apache Camel

PS C:\Users\p3060630> iex "& { $(iwr https://ps.jbang.dev) } app install --verbose --fresh --force camel@apache/camel"
WARNING: Windows Developer Mode is not enabled on your system, this is necessary
for JBang to be able to function correctly, see this page for more information:
https://www.jbang.dev/documentation/guide/latest/usage.html#usage-on-windows
WARNING: Windows Developer Mode is not enabled on your system, this is necessary
for JBang to be able to function correctly, see this page for more information:
https://www.jbang.dev/documentation/guide/latest/usage.html#usage-on-windows
[jbang] [0:707] jbang version 0.111.0
[jbang] [0:727] Resolving resource ref: camel@apache/camel
[jbang] [0:746] Reading catalog from C:\Users\p3060630\.jbang\implicit-catalog.json
[jbang] [0:840] Reading catalog from C:\Users\p3060630\.jbang\implicit-catalog.json
[jbang] [1:185] Requesting HTTP GET https://www.jbang.dev/releases/latest/download/version.txt
[jbang] [1:185] Requesting HTTP GET https://raw.githubusercontent.com/apache/camel/HEAD/jbang-catalog.json
[jbang] [1:186] Headers {User-Agent=[JBang/0.111.0 (Windows 10/10.0/amd64) Java/17.0.8/Oracle Corporation]}
[jbang] [1:187] Headers {User-Agent=[JBang/0.111.0 (Windows 10/10.0/amd64) Java/17.0.8/Oracle Corporation]}
[jbang] [1:525] [ERROR] Could not download https://github.com/apache/camel/blob/HEAD/jbang-catalog.json
dev.jbang.cli.ExitException: Could not download https://github.com/apache/camel/blob/HEAD/jbang-catalog.json
        at dev.jbang.source.resolvers.RemoteResourceResolver.fetchFromURL(RemoteResourceResolver.java:118)
        at dev.jbang.source.resolvers.RemoteResourceResolver.resolve(RemoteResourceResolver.java:42)
        at dev.jbang.source.resolvers.CombinedResourceResolver.lambda$resolve$0(CombinedResourceResolver.java:27)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:1002)
        at dev.jbang.Main.main(Main.java:14)
Caused by: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

**Comment by maxandersen** [comment]
Do you have a proxy that modifies certificates for unapproved sites?

**Reply by nmvprasad** [reply]
No Proxy and I was able to download those jars independently, only it is giving error in karavan docker
![image](https://github.com/jbangdev/jbang/assets/20818508/a68d4ec3-b40e-4de8-9f4d-918b2739d245)

I try to follow these steps to install and run Karavan on my local using docker 
https://github.com/apache/camel-karavan/blob/main/docs/WEB_DOCKER.md

looks like karavan is build with OpenJDK java image and it is not able to download due certificate from this mavan repository.
So not sure how to add this certificate or force to sue different jdk which is there on my local for karavan

https://repo1.maven.org/maven2): PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

Docker CL:
# java -version
openjdk version "17.0.8.1" 2023-08-24
OpenJDK Runtime Environment Temurin-17.0.8.1+1 (build 17.0.8.1+1)
OpenJDK 64-Bit Server VM Temurin-17.0.8.1+1 (build 17.0.8.1+1, mixed mode, sharing)

# echo $JAVA_HOME                   
/opt/java/openjdk
# ls /opt/java/openjdk
bin  conf  include  jmods  legal  lib  man  NOTICE  release
# cd lib
# ls
apt  compat-ld  dpkg  gold-ld  init  locale  lsb  mime  os-release  sasl2  ssl  sysctl.d  systemd  terminfo  tmpfiles.d  udev  usrmerge  x86_64-linux-gnu

**Reply by maxandersen** [reply]
Sounds Weird if karavan is somehow using image that cant reach maven central. 

Maybe @davsclaus has an idea?

**Reply by maxandersen** [reply]
See https://stackoverflow.com/questions/48593183/maven-unable-to-find-valid-certification-path#comment135013686_73927521

**Comment by davsclaus** [comment]
It seems it cannot download `https://github.com/apache/camel/blob/HEAD/jbang-catalog.json` which I wonder if the user is behind some corporate HTTP proxy or something?

**Reply by maxandersen** [reply]
That is and was my first thought too but last comment points to the difference being the karavan image but still seems unlikely there isn't some network layer messing with the certificates

**Reply by davsclaus** [reply]
Yes and networking inside docker and running on host os like windows et all can cause pain. Its a more general docker problem and you should google the internet for others with such problems and see how they fixed it

**Comment by davsclaus** [comment]
@mgubaidullin wonder if you have seen this before ^^^
